packages/rsx/src/node.rs

```diff
@@ -112,7 +112,27 @@ impl Parse for BodyNode {
             return Ok(BodyNode::IfChain(stream.parse()?));
         }
 
-        Ok(BodyNode::RawExpr(stream.parse::<Expr>()?))
+        // Match statements are special but have no special arm syntax
+        // we could allow arm syntax if we wanted
+        //
+        // ```
+        // match {
+        //  val => div {}
+        //  other_val => div {}
```

Comment:
I think allowing you to do normal rsx in match arms would be more consistant with if statements and loops in the future, but it doesn't need to be in this PR